profiler/test/Datadog.Profiler.IntegrationTests/Contention/ContentionProfilerTest.cs

```diff
@@ -0,0 +1,113 @@
+        public void ShouldContentionProfilerBeDisabledByDefault(string appName, string framework, string appAssembly)
+        {
+            var runner = new TestApplicationRunner(appName, framework, appAssembly, _output, commandLine: ScenarioContention);
+            runner.Environment.SetVariable(EnvironmentVariables.WallTimeProfilerEnabled, "0");
+
+            using var agent = new MockDatadogAgent(_output);
+
+            runner.Run(agent);
+
+            ExtractContentionSamples(runner.Environment.PprofDir).Should().BeEmpty();
+        }
+
+        [TestAppFact("Samples.Computer01", new[] { "net6.0" })]
+        public void ExplicitlyDisableContentionProfiler(string appName, string framework, string appAssembly)
+        {
+            var runner = new TestApplicationRunner(appName, framework, appAssembly, _output, commandLine: ScenarioContention);
+
+            runner.Environment.SetVariable(EnvironmentVariables.ContentionProfilerEnabled, "0");
+
+            using var agent = new MockDatadogAgent(_output);
+
+            runner.Run(agent);
+
+            ExtractContentionSamples(runner.Environment.PprofDir).Should().BeEmpty();
+        }
+
+        private static IEnumerable<(long Count, long Duration, StackTrace Stacktrace)> ExtractContentionSamples(string directory)
+        {
+            static IEnumerable<(long Count, long Duration, StackTrace Stacktrace, long Time)> GetContentionSamples(string directory)
+            {
+                foreach (var file in Directory.EnumerateFiles(directory, "*.pprof", SearchOption.AllDirectories))
+                {
+                    using var stream = File.OpenRead(file);
+                    var profile = Profile.Parser.ParseFrom(stream);
+
+                    foreach (var sample in profile.Sample)
+                    {
+                        var count = sample.Value[ContentionCountSlot];
+                        if (count == 0)
+                        {
+                            continue;
+                        }
+
+                        var size = sample.Value[ContentionDurationSlot];
+
+                        var labels = sample.Labels(profile).ToArray();
+
+                        yield return (count, size, sample.StackTrace(profile), profile.TimeNanos);
+                    }
+                }
+            }
+
+            return GetContentionSamples(directory)
+                .OrderBy(s => s.Time)
+                .Select(s => (s.Count, s.Duration, s.Stacktrace));
+        }
+
+        private void CheckContentionProfiles(TestApplicationRunner runner)
+        {
+            using var agent = new MockDatadogAgent(_output);
+
+            runner.Run(agent);
+
+            Assert.True(agent.NbCallsOnProfilingEndpoint > 0);
+
+            var contentionSamples = ExtractContentionSamples(runner.Environment.PprofDir).ToArray();
+            contentionSamples.Should().NotBeEmpty();
+
+            Assert.All(contentionSamples,  s => Assert.True(s.Duration > 0));
```

Comment:
The Assert should be made on Count instead of Duration since this is what GetContentionSamples() does.
I'm not sure if Duration could be 0 in case of very fast contention...

Reply:
as you want, checking Count or Duration it's the same to me

Reply:
there is really little chance for the duration to be equal to 0, since it's was spinning before...

Reply:
✔️